tests/instrumentation/asyncio/asyncpg_tests.py

```diff
@@ -0,0 +1,114 @@
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import pytest
+
+from elasticapm.conf import constants
+
+asyncpg = pytest.importorskip("asyncpg")  # isort:skip
+pytestmark = [pytest.mark.asyncpg, pytest.mark.asyncio]
+
+if "POSTGRES_DB" not in os.environ:
+    pytestmark.append(
+        pytest.mark.skip(
+            "Skipping asyncpg tests, no POSTGRES_DB environment variable set"
+        )
+    )
+
+
+def dsn():
+    return "postgres://{user}:{password}@{host}:{port}/{database}".format(
+        **{
+            "database": os.environ.get("POSTGRES_DB", "elasticapm_test"),
+            "user": os.environ.get("POSTGRES_USER", "postgres"),
+            "password": os.environ.get("POSTGRES_PASSWORD", "postgres"),
+            "host": os.environ.get("POSTGRES_HOST", "localhost"),
+            "port": os.environ.get("POSTGRES_PORT", "5432"),
+        }
+    )
+
+
+@pytest.fixture()
+async def connection(request):
+    conn = await asyncpg.connect(dsn())
+
+    # we use a finalizer instead of yield, because Python 3.5 throws
+    # a syntax error, even if the test doesn't run
+    def rollback():
+        conn.execute("ROLLBACK")
+        conn.close()
+
+    request.addfinalizer(rollback)
+    await conn.execute(
+        "BEGIN;"
+        "CREATE TABLE test(id int, name VARCHAR(5) NOT NULL);"
+        "INSERT INTO test VALUES (1, 'one'), (2, 'two'), (3, 'three');"
+    )
+    return conn
+
+
+async def test_execute_with_sleep(instrument, connection, elasticapm_client):
+    elasticapm_client.begin_transaction("test")
+    await connection.execute("SELECT pg_sleep(0.1);")
+    elasticapm_client.end_transaction("test", "OK")
+
+    transaction = elasticapm_client.events[constants.TRANSACTION][0]
+    spans = elasticapm_client.spans_for_transaction(transaction)
+
+    assert len(spans) == 1
+    span = spans[0]
+    assert 100 < span["duration"] < 110
+    assert transaction["id"] == span["transaction_id"]
+    assert span["type"] == "db"
+    assert span["subtype"] == "postgres"
+    assert span["action"] == "query"
+    assert span["sync"] == False
+    assert span["name"] == "SELECT FROM"
+
+
+async def test_executemany(instrument, connection, elasticapm_client):
+    elasticapm_client.begin_transaction("test")
+    await connection.executemany(
+        "INSERT INTO test VALUES ($1, $2, $3);",
+        [(1, "winter"), (2, "spring"), (3, "summer")],
+    )
+    elasticapm_client.end_transaction("test", "OK")
+
+    transaction = elasticapm_client.events[constants.TRANSACTION][0]
+    spans = elasticapm_client.spans_for_transaction(transaction)
+
+    assert len(spans) == 1
+    span = spans[0]
+    assert transaction["id"] == span["transaction_id"]
+    assert span["subtype"] == "postgres"
+    assert span["action"] == "query"
+    assert span["sync"] == False
+    assert span["name"] == "SELECT FROM test"
```

Comment:
This does not seem like desired behavior. We've executed an insert but the report says `SELECT`

Reply:
great catch. thanks!